From: Stephen Rothwell <sfr@canb.auug.org.au>
Subject: writeback-support-retrieving-per-group-debug-writeback-stats-of-bdi-fix-2
Date: Tue, 23 Apr 2024 10:22:53 +1000

fix build with CONFIG_CGROUP_WRITEBACK=n

mm/backing-dev.c: In function 'wb_stats_show':
mm/backing-dev.c:175:33: error: 'struct bdi_writeback' has no member named =
'memcg_css'
  175 |                    cgroup_ino(wb->memcg_css->cgroup),
      |                                 ^~

Link: https://lkml.kernel.org/r/20240423102253.3675d0af@canb.auug.org.au
Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
Cc: Kemeng Shi <shikemeng@huaweicloud.com>
Cc: Tejun Heo <tj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/backing-dev.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/mm/backing-dev.c~writeback-support-retrieving-per-group-debug-writeback-stats-of-bdi-fix-2
+++ a/mm/backing-dev.c
@@ -172,7 +172,11 @@ static void wb_stats_show(struct seq_fil
 		   "b_more_io:         %10lu\n"
 		   "b_dirty_time:      %10lu\n"
 		   "state:             %10lx\n\n",
+#ifdef CONFIG_CGROUP_WRITEBACK
 		   cgroup_ino(wb->memcg_css->cgroup),
+#else
+		   (ino_t)0,
+#endif
 		   K(stats->nr_writeback),
 		   K(stats->nr_reclaimable),
 		   K(stats->wb_thresh),
_
